firedrake/linear_solver.py

```diff
@@ -175,5 +176,6 @@ def solve(self, x, b):
             raise ConvergenceError("LinearSolver failed to converge after %d iterations with reason: %s", self.ksp.getIterationNumber(), solving_utils.KSPReasons[r])
 
         # Grab the comm associated with `x` and call PETSc's garbage cleanup routine
+        gc.collect()
         comm = x.function_space().mesh()._comm
         PETSc.garbage_cleanup(comm=comm)
```

Comment:
I think we can just call the petsc4py routine here, since we know exactly what communicator we want to clean up. I also think calling `gc.collect()` here might be really expensive and it might need to be removed.

If a user calls `garbage_cleanup()` I don't care if it's slow. The user might...

Reply:
I vote that `gc.collect()` be removed. It's an expensive call and no one should notice since it happens internally. We definitely need it for the public cleanup interface, as otherwise users might think it's a bug, but here it can go.